File: backend/src/main/java/com/gestionstock/backend/service/fournisseur/FournisseurService.java
package com.gestionstock.backend.service.fournisseur;

import com.gestionstock.backend.dto.fournisseur.FournisseurDTO;

import java.util.List;

/**
 * Interface du service Fournisseur.
 * Définit les opérations métier pour la gestion des fournisseurs.
 */
public interface FournisseurService {

    /** Récupère tous les fournisseurs */
    List<FournisseurDTO> getAllFournisseurs();

    /** Récupère un fournisseur par son identifiant */
    FournisseurDTO getFournisseurById(Long id);

    /** Recherche les fournisseurs par nom (recherche partielle) */
    List<FournisseurDTO> searchFournisseurs(String nom);

    /** Crée un nouveau fournisseur */
    FournisseurDTO createFournisseur(FournisseurDTO fournisseurDTO);

    /** Met à jour un fournisseur existant */
    FournisseurDTO updateFournisseur(Long id, FournisseurDTO fournisseurDTO);

    /** Supprime un fournisseur par son identifiant */
    void deleteFournisseur(Long id);
}

File: frontend/src/app/core/services/fournisseur.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Fournisseur } from '../models/fournisseur.model';

const API_URL = 'http://localhost:8080/api';

/**
 * Service Angular pour la gestion des fournisseurs.
 * Communique avec le backend via les endpoints REST /api/fournisseurs.
 */
@Injectable({
  providedIn: 'root'
})
export class FournisseurService {

  private http = inject(HttpClient);

  /** Récupère tous les fournisseurs */
  getAllFournisseurs(): Observable<Fournisseur[]> {
    return this.http.get<Fournisseur[]>(`${API_URL}/fournisseurs`);
  }

  /** Récupère un fournisseur par son ID */
  getFournisseurById(id: number): Observable<Fournisseur> {
    return this.http.get<Fournisseur>(`${API_URL}/fournisseurs/${id}`);
  }

  /** Recherche les fournisseurs par nom */
  searchFournisseurs(nom: string): Observable<Fournisseur[]> {
    return this.http.get<Fournisseur[]>(`${API_URL}/fournisseurs/search?nom=${nom}`);
  }

  /** Crée un nouveau fournisseur */
  createFournisseur(fournisseur: Fournisseur): Observable<Fournisseur> {
    return this.http.post<Fournisseur>(`${API_URL}/fournisseurs`, fournisseur);
  }

  /** Met à jour un fournisseur existant */
  updateFournisseur(id: number, fournisseur: Fournisseur): Observable<Fournisseur> {
    return this.http.put<Fournisseur>(`${API_URL}/fournisseurs/${id}`, fournisseur);
  }

  /** Supprime un fournisseur */
  deleteFournisseur(id: number): Observable<void> {
    return this.http.delete<void>(`${API_URL}/fournisseurs/${id}`);
  }
}
